content/en/docs/languages/java/configuration.md

```diff
@@ -0,0 +1,413 @@
+- [io.opentelemetry.contrib:opentelemetry-gcp-resources](https://github.com/open-telemetry/opentelemetry-java-contrib/tree/main/gcp-resources)
+  includes providers for
+  [common GCP resources](https://github.com/open-telemetry/opentelemetry-java-contrib/tree/main/gcp-resources/src/main/java/io/opentelemetry/contrib/gcp/resource)
+
+### Disabling automatic ResourceProviders
+
+If you are using the `ResourceProvider` SPI, which many instrumentation agent
+distributions include automatically, you can turn on or off one or more of them
+by using the following configuration items:
+
+| Environment variable                    | Description                                                                                  |
+| --------------------------------------- | -------------------------------------------------------------------------------------------- |
+| `otel.java.enabled.resource.providers`  | Turns on one or more `ResourceProvider` types. If unset, all resource providers are enabled. |
+| `otel.java.disabled.resource.providers` | Turns off one or more `ResourceProvider` types.                                              |
+
+The value for these properties must be a comma-separated list of fully qualified
+`ResourceProvider` class names. For example, if you don't want to expose the
+name of the operating system through the resource, you can pass the following
+JVM argument:
+
+`-Dotel.java.disabled.resource.providers=io.opentelemetry.instrumentation.resources.OsResourceProvider`
+
+### Attribute limits
+
+These properties can be used to control the maximum number and length of
+attributes.
+
+| System property                     | Description                                                                            | Default  |
+| ----------------------------------- | -------------------------------------------------------------------------------------- | -------- |
+| `otel.attribute.value.length.limit` | The maximum length of attribute values. Applies to spans and logs.                     | No limit |
+| `otel.attribute.count.limit`        | The maximum number of attributes. Applies to spans, span events, span links, and logs. | `128`    |
+
+### Propagators
+
+The propagators determine which distributed tracing header formats are used, and
+which baggage propagation header formats are used.
+
+| System property    | Description                                                                      | Default                      |
+| ------------------ | -------------------------------------------------------------------------------- | ---------------------------- |
+| `otel.propagators` | The propagators to be used. Use a comma-separated list for multiple propagators. | `tracecontext,baggage` (W3C) |
+
+Supported values are the following:
+
+| Value          | Description                                                                                                      |
```

Comment:
Can we add a column to this table with the artifact that contains each of these?

- `tracecontext` lives in `opentelemetry-api`
- `baggage` lives in `opentelemetry-api`
- `b3` lives in `opentelemetry-extension-trace-propagators`
- `b3multi` lives in `opentelemetry-extension-trace-propagators`
- `jaeger` lives in `opentelemetry-extension-trace-propagators`
- `xray` lives in `io.opentelemetry.contrib:opentelemetry-aws-xray-propagator`
- `ottrace` lives in `opentelemetry-extension-trace-propagators`